src/tests/ftest/launch.py

```diff
@@ -371,13 +456,105 @@ def get_test_files(test_list, args, tmp_dir):
     return test_files
 
 
+def get_nvme_replacement(args):
+    """Determine the value to use for the '--nvme' command line argument.
+
+    Parse the lspci output for any NMVe devices, e.g.
+        $ lspci | grep 'Non-Volatile memory controller:'
+        5e:00.0 Non-Volatile memory controller:
+            Intel Corporation NVMe Datacenter SSD [3DNAND, Beta Rock Controller]
+        5f:00.0 Non-Volatile memory controller:
+            Intel Corporation NVMe Datacenter SSD [3DNAND, Beta Rock Controller]
+        81:00.0 Non-Volatile memory controller:
+            Intel Corporation NVMe Datacenter SSD [Optane]
+        da:00.0 Non-Volatile memory controller:
+            Intel Corporation NVMe Datacenter SSD [Optane]
+
+    Optionally filter the above output even further with a specified search
+    string (e.g. '--nvme=auto:Optane'):
+        $ lspci | grep 'Non-Volatile memory controller:' | grep 'Optane'
+        81:00.0 Non-Volatile memory controller:
+            Intel Corporation NVMe Datacenter SSD [Optane]
+        da:00.0 Non-Volatile memory controller:
+            Intel Corporation NVMe Datacenter SSD [Optane]
+
+    Args:
+        args (argparse.Namespace): command line arguments for this program
+
+    Returns:
+        str: a comma-separated list of nvme device pci addresses available on
+            all of the specified test servers
+
+    """
+    # A list of server host is required to able to auto-detect NVMe devices
+    if not args.test_servers:
+        print("ERROR: Missing a test_servers list to auto-detect NVMe devices")
+        exit(1)
+
+    # Get a list of NVMe devices from each specified server host
+    host_list = args.test_servers.split(",")
+    command_list = [
+        "/usr/sbin/lspci -D", "grep 'Non-Volatile memory controller:'"]
+    if ":" in args.nvme:
+        command_list.append("grep '{}'".format(args.nvme.split(":")[1]))
+    command = " | ".join(command_list)
+    task = get_remote_output(host_list, command)
+
+    # Verify the command was successful on each server host
+    if not check_remote_output(task, command):
+        print("ERROR: Issuing commands to detect NVMe PCI addresses.")
+        exit(1)
+
+    # Verify each server host has the same NVMe PCI addresses
+    output_data = list(task.iter_buffers())
+    if len(output_data) > 1:
+        print("ERROR: Non-homogeneous NVMe PCI addresses.")
+        exit(1)
```

Comment:
This is just short-term, yes?  Ultimately we need to be able to support heterogeneous clusters, yes?  Of course that will be at a time when we are creating per-server configs and not expecting a single config to work for every node.

Reply:
Yes, short term - while we still use a single config file for every node.